src/api/spec/jobs/unstage_requests_job_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe UnstageRequestsJob, type: :job, vcr: true do
+  include ActiveJob::TestHelper
+
+  describe '#perform' do
+    let(:user) { create(:confirmed_user) }
+    let(:project) { create(:project_with_package) }
+    let(:staging_workflow) { create(:staging_workflow_with_staging_projects, project: project) }
+    let(:staging_project) { staging_workflow.staging_projects.first }
+    let(:source_project) { create(:project, name: 'source_project') }
+    let(:target_package) { create(:package, name: 'target_package', project: project) }
+    let(:source_package) { create(:package, name: 'source_package', project: source_project) }
+    let(:bs_request) do
+      create(:bs_request_with_submit_action,
+             target_project: project.name,
+             target_package: target_package.name,
+             source_project: source_project.name,
+             source_package: source_package.name)
+    end
+
+    # assign request to staging_project
+    before do
+      bs_request.staging_project = staging_project
+      bs_request.save
+    end
+
+    subject! { UnstageRequestsJob.new.perform(staging_project, [bs_request.id], user) }
+
+    it { expect { subject }.not_to raise_error }
```

Comment:
Maybe we could also check that branch object does exist afterwards? Not sure how this works with testing jobs though..

Something like 
```
it { expect { Package.find_by_project_and_name(staging_project.name, source_package.name) }.to be_present }
```
maybe?